fix: Use `ts-node` rather than bash for scripting integration bundling

timfish opened the pull request:
I was using my Windows machine for the first time in ages and noticed this remaining bash script. 

This PR converts the script to TypeScript and calls `ts-node`

AbhiPrasad commented on packages/integrations/scripts/buildBundles.ts:
```diff
@@ -0,0 +1,22 @@
+  // The index file is only there for the purposes of npm builds (for the CDN we create a separate bundle for each
+  // integration) and the flags file is just a helper for including or not including debug logging, whose contents gets
+  // incorporated into each of the individual integration bundles, so we can skip them both here.
+  return readdirSync(join(__dirname, '..', 'src')).filter(file => !file.endsWith('index.ts'));
```
isn't the shell script using a `==` check instead of a substring check? Why did we do `endsWith` here?

Also nit: would be nice to extract these into vars so you know all the return values.

```ts
const fileNames = join(__dirname, '..', 'src');
const files = readdirSync(fileNames);
return files.filter(...);
```

AbhiPrasad commented on packages/integrations/scripts/buildBundles.ts:
```diff
@@ -0,0 +1,22 @@
+}
+
+for (const integration of getIntegrations()) {
+  for (const jsVersion of ['ES5', 'ES6']) {
```
I wonder if we can run this async on both es5 and es6 at the same time.

timfish replied:
We probably also need to check for non-zero exit code too...

AbhiPrasad replied:
yeah - thats probably the tricky part. Would be interesting to try out here!

timfish replied:
Oh, its about 5x quicker!

AbhiPrasad replied:
alright I'm in favour of shipping that - but we'll need to make sure to have some kind of flag that can bring back the sync behavior if we need to debug.

timfish commented:
Builds are now run in parallel. 

On this machine previously `yarn build` in integrations took around 60 seconds. Now it completes in roughly 10s.

timfish commented:
Oh no! I had the `!== 'index.ts'`  logic inverted so it was only building an integration for `index.ts` 🤦‍♂️

Now I've fixed this and added a `--parallel` flag we have:
- Sequential - 55s
- Parallel - 27s

So it's still about 2x as quick on this machine.

lobsterkatie reviewed (approved):
This has long been on my mental backlog - thanks for finally getting it done!
